docs/docs/docs/01-core/data-transfer/01-engine.md

```diff
@@ -0,0 +1,304 @@
+`'patch'` - require only the semver major, minor, and patch version to match (but allow tag differences such as -alpha and -beta)
+
+The default `versionStrategy` used when one is not provided is `'ignore'`.
+
+#### schemaStrategy
+
+The follow `schemaStrategy` values may be used:
+
+`'ignore'` - bypass schema validation (transfer will attempt to run but throw errors on incompatible data type inserts)
+`'strict'` - disallow mismatches that are expected to cause errors in the transfer, but allow certain non-data fields in the schema to differ
+`'exact'` - schema must be identical with no changes
+
+Note: The "strict" schema strategy is defined as "anything expected to cause errors in the transfer" and is the default method for the import, export, and transfer CLI commands. Therefore, the technical functionality will always be subject to change. If you need to find the definition for the current version of Strapi, see `packages/core/data-transfer/src/engine/validation/schemas/index.ts`
+
+The default `schemaStrategy` used when one is not provided is `'strict'`.
+
+##### Handling Schema differences
+
+When a schema diff is discovered with a given schemaStrategy, an error is throw. However, before throwing the error the engine checks to see if there are any schema diff handlers set via `engine.onSchemaDiff(handler)` which allows errors to be bypassed (for example, by prompting the user if they wish to proceed).
+
+A diff handler is an optionally asynchronous middleware function that accepts a `context` and a `next` parameter.
+
+`context` is an object of type `SchemaDiffHandlerContext`
+
+```typescript
+// type Diff can be found in /packages/core/data-transfer/src/utils/json.ts
+type SchemaDiffHandlerContext = {
+  ignoredDiffs: Record<string, Diff[]>;
+  diffs: Record<string, Diff[]>;
+  source: ISourceProvider;
+  destination: IDestinationProvider;
+};
+```
+
+`next` is a function that is called, passing the modified `context` object, to proceed to the next middleware function.
+
+```typescript
+const diffHandler = async (context, next) => {
+  const ignoreThese = {};
+  // loop through the diffs
+  Object.entries(context.diffs).forEach(([uid, diffs]) => {
+    for (const [i, diff] of diffs) {
+      // get the path of the diff in the schema
+      const path = [uid].concat(diff.path).join('.');
+
+      // Allow a diff on the country schema displayName
+      if (path === 'api::country.country.info.displayName') {
+        if (!isArray(context.ignoredDiffs[uid])) {
+          context.ignoredDiffs[uid] = [];
+        }
+        context.ignoredDiffs[uid][i] = diff;
+      }
+    }
+  });
+
+  return next(context);
+};
+
+engine.onSchemaDiff(diffHandler);
+```
+
+After all the schemaDiffHandler middleware has been run, another diff is run between `context.ignoredDiffs` and `context.diffs` and any remaining diffs that have not been ignored are thrown as fatal errors and the engine will abort the transfer.
```

Comment:
```suggestion
After all the schemaDiffHandler middlewares have been run, another diff is run between `context.ignoredDiffs` and `context.diffs` and any remaining diffs that have not been ignored are thrown as fatal errors and the engine will abort the transfer.
```

Reply:
I believe that's actually correct because English is weird. But if you think it's confusing I could make it "middlware functions"

Reply:
Damn it, for once I thought I got you :sweat_smile: 
If it's intentional I fully trust you